xrootd/cmd/xrootd-fuse/main.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Command xrootd-fuse mounts the directory contents of a remote xrootd server to a local directory.
+//
+// Usage:
+//
+//  $> xrootd-fuse [OPTIONS] <remote-dir> <local-dir>
+//
+// Example:
+//
+//  $> xrootd-fuse root://server.example.com/some/dir /mnt
+//  $> xrootd-fuse -debug=true root://server.example.com/some/dir /mnt
+//
+// Options:
+//   -debug
+//         print debugging messages.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hanwen/go-fuse/fuse/nodefs"
+	"github.com/hanwen/go-fuse/fuse/pathfs"
+	"go-hep.org/x/hep/xrootd/client"
+	"go-hep.org/x/hep/xrootd/xrdfuse"
+	"go-hep.org/x/hep/xrootd/xrdio"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `xrootd-fuse mounts the directory contents of a remote xrootd server to a local directory.
+
+Usage:
+
+ $> xrootd-fuse [OPTIONS] <remote-dir> <local-dir>
+
+Example:
+
+ $> xrootd-fuse root://server.example.com/some/dir /mnt
+ $> xrootd-fuse -debug=true root://server.example.com/some/dir /mnt
+
+Options:
+`)
+		flag.PrintDefaults()
+	}
+}
+
+func main() {
+	log.SetPrefix("xrootd-fuse: ")
+	log.SetFlags(0)
+
+	debug := flag.Bool("debug", false, "print debugging messages.")
+
+	flag.Parse()
+
+	if flag.NArg() != 2 {
+		flag.Usage()
+		log.Fatalf("missing directory operands")
+	}
+
+	url, err := xrdio.Parse(flag.Arg(0))
+
+	c, err := client.NewClient(context.Background(), url.Addr, url.User)
+	if err != nil {
+		fmt.Printf("Connection fail: %v\n", err)
+		os.Exit(1)
+
+	}
+
+	fs := xrdfuse.NewFS(c, url.Path, func(e error) {
+		log.Println(e)
+	})
+	nfs := pathfs.NewPathNodeFs(fs, nil)
+	server, _, err := nodefs.MountRoot(flag.Arg(1), nfs.Root(), &nodefs.Options{
+		Debug: *debug,
+	})
+	if err != nil {
+		fmt.Printf("Mount fail: %v\n", err)
+		os.Exit(1)
+	}
+
+	server.Serve()
```

Comment:
we should probably arrange things to properly unmount the filesystem when the program receives `^C`.
kind of what I did in the `rootio` FUSE command:
https://github.com/go-hep/hep/pull/223/files#diff-9884b867f20406412cddcbbf7413f557R19